ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -746,6 +746,11 @@ default void validatePartSpec(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
     throw new UnsupportedOperationException("Storage handler does not support validation of partition values");
   }
 
+  default void validatePartAnySpec(org.apache.hadoop.hive.ql.metadata.Table hmsTable, Map<String, String> partitionSpec)
```

Comment:
what does it do?

Reply:
Validates that the provided partitionSpec is a valid according to the current or past table partitioning.
The existing method validatePartSpec validates that the spec is valid based on current table partitioning only. This isn't enough for Iceberg compaction on partition level with evolution because the partition spec provided in compaction command may correspond for an old partitioning.

Reply:
i would keep the same name since we overload the method with extra param + we should add a javadoc

Reply:
I have already added a Javadoc.
The method is not overloaded, it has same params as existing method.